garak/probes/base.py

```diff
@@ -139,54 +139,62 @@ def _execute_attempt(self, this_attempt):
         if self.post_buff_hook:
             this_attempt = self._postprocess_buff(this_attempt)
         this_attempt = self._postprocess_hook(this_attempt)
-        _config.transient.reportfile.write(json.dumps(this_attempt.as_dict()) + "\n")
         self._generator_cleanup()
         return copy.deepcopy(this_attempt)
 
-    def probe(self, generator) -> Iterable[garak.attempt.Attempt]:
-        """attempt to exploit the target generator, returning a list of results"""
-        logging.debug("probe execute: %s", self)
-
-        self.generator = generator
-
-        # build list of attempts
-        attempts_todo: Iterable[garak.attempt.Attempt] = []
-        prompts = list(self.prompts)
-        for seq, prompt in enumerate(prompts):
-            attempts_todo.append(self._mint_attempt(prompt, seq))
-
-        # buff hook
-        if len(_config.buffmanager.buffs) > 0:
-            attempts_todo = self._buff_hook(attempts_todo)
-
-        # iterate through attempts
+    def _execute_all(self, attempts) -> Iterable[garak.attempt.Attempt]:
```

Comment:
can `docs/source/garak.probes.base.rst` get a small update to document this change?

Reply:
New commit c5eaf39 adds doc changes, I am not 100% sure on formatting or content so please offer any suggestions you believe are needed still.

Reply:
I think this is fine!